### PR TITLE
tests: added old OpenSSL client to test suite

### PR DESCRIPTION
**Pull request by spacewander**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-resty-core project.

### CODE REVIEW

File: t/ssl.t [hunk 1]
```diff
@@ -2168,3 +2170,84 @@ client ip: 127.0.0.1
 [error]
 [alert]
 [emerg]
+
+
+
+=== TEST 21: yield during doing handshake with client which uses low version OpenSSL
```
Comment (agentzh) [inline t/ssl.t]: Adding `--- ONLY` to this test case and running this test file yields:

```
/bin/openssl
OpenSSL 1.0.2m-fips  2 Nov 2017
t/ssl.t .. # I found ONLY: maybe you're debugging?
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] still could not bind()
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] still could not bind()
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:4443 failed (98: Address already in use)
```

Running other test blocks individually in this test file does not have any problems.

File: t/ssl-session-fetch.t [hunk 2]
```diff
@@ -398,3 +400,94 @@ $/s,
 [alert]
 [emerg]
 [error]
+
+
+
+=== TEST 5: yield during doing handshake with client which uses low version OpenSSL
```
[the remark this answers is not in the document]
Comment (spacewander) [reply t/ssl-session-fetch.t]: @agentzh 
I could not reproduce it with both OpenSSL 0.9.8o and OpenSSL 1.0.2m in my machine. And I could not reproduce it in travis ci: https://travis-ci.org/spacewander/lua-resty-core/builds/327085669

File: t/ssl-session-fetch.t [hunk 3]
```diff
@@ -398,3 +400,94 @@ $/s,
...
+        ssl_certificate_key $TEST_NGINX_CERT_DIR/cert/test.key;
+    }
+--- config
+    lua_ssl_trusted_certificate $TEST_NGINX_CERT_DIR/cert/test.crt;
+
+    location /t {
+        set $sess_file $TEST_NGINX_HTML_DIR/sess;
+        set $addr 127.0.0.1:$TEST_NGINX_SERVER_SSL_PORT;
+        content_by_lua_block {
+            do
```
Comment (membphis) [inline t/ssl-session-fetch.t]: we can remove this `do ... end`

File: t/ssl-session-fetch.t (hunk 2 above)
[the remark this answers is not in the document]
Comment (agentzh) [reply t/ssl-session-fetch.t]: @spacewander Did you try `--- ONLY`? I can reproduce it 100% on my side.

[the remark this answers is not in the document]
Comment (agentzh) [reply t/ssl-session-fetch.t]: @spacewander It's not related to the OpenSSL version I think.

[the remark this answers is not in the document]
Comment (spacewander) [reply t/ssl-session-fetch.t]: @agentzh 
Yes, I try it with `--- ONLY` in my machine and travis build. It seems to be platform dependency(or machine dependency if worse).

**Comment by spacewander** [comment]
@membphis @agentzh
A new commit is pushed to address issues found in review.

**Pull request closed without merging**